File: files/arc_openmp/fortran/hello.f90
program hello_world

    ! Load the OpenMP functions library
    use omp_lib

    implicit none

    integer :: nthreads, tid

    !  Fork a team of threads, with private versions of the declared variables.
    !$OMP PARALLEL PRIVATE(nthreads, tid)

    ! Get the thread number and print it 
    tid = OMP_GET_THREAD_NUM()
    print *, "Hello World from thread number ", tid

    ! Only the master thread does the following
    if (tid == 0) then
        nthreads = OMP_GET_NUM_THREADS()
        print *, "Number of threads = ", nthreads
    end if

    ! Disband all but the master thread
    !$OMP END PARALLEL

end program hello_world

File: files/arc_openmp/c/hello.c
// Load the OpenMP functions library
#include<omp.h>

int main()
  {
    int nthreads, tid;
    // Fork a team of threads, with private versions of the declared variables.
    // #pragma omp parallel private(nthreads, tid)
    // {
    // Get the thread number and print it 
    tid = omp_get_thread_num();
    printf("Hello World from thread number %d\n", tid);

    // Only the master thread does the following
    // #pragma omp barrier
    // #pragma omp master
    // {    
    if (tid == 0)
      {
	nthreads = omp_get_num_threads();
	printf("Number of threads = %d\n", nthreads);
      }
    // }
    
    // End of pragma, disband all but the master thread
    //}
    return 0;
  }
